services/jwk.py

```diff
@@ -39,6 +42,20 @@ def validate_admin(token) -> bool:
             return False
         return True
 
+    @staticmethod
+    def validate_role(token, valid_roles) -> bool:
+        try:
+            decoded = jwt.decode(token, __secret__, algorithms=["HS256"])
+            user_id = decoded.get("sub")
+            with session_scope() as session:
+                users = session.query(User.id.label("ID"), User.role.label('role')).filter(User.id == user_id)
```

Comment:
This will be quite expensive. I would suggest adding this to the jwt (generate on line 15) when the user signs in. You should just be able to pass it as a parameter and add another key value to the jwt. I believe that both login and register already have access to role, so it shouldn't be much extra work.